src/app/ui-components/selectFeeRate/index.tsx

```diff
@@ -138,20 +142,38 @@ function SelectFeeRate({
             {t('COMMON.EDIT')} <PencilSimple size={16} weight="fill" />
           </Label>
         </EditRow>
-        <Label $size="s" $variant="dark">
-          <NumericFormat
-            value={feeRate}
-            displayType="text"
-            thousandSeparator
-            renderText={(value: string) => (
-              <>
-                {value} {feeRateUnits}
-              </>
-            )}
-          />
-        </Label>
+        {feeRateUnits && (
+          <Label $size="s" $variant="dark">
+            <NumericFormat
+              value={feeRate}
+              displayType="text"
+              thousandSeparator
+              renderText={(value: string) => (
+                <>
+                  {value} {feeRateUnits}
+                </>
+              )}
+            />
+          </Label>
+        )}
+        {fee && !feeRateUnits && (
+          <RowContainer>
+            <div />
+            <NumericFormat
+              value={baseToFiat(fee)}
+              displayType="text"
+              prefix={`~ ${currencySymbolMap[fiatUnit]}`}
+              thousandSeparator
+              renderText={(value: string) => (
+                <Label $size="s" $variant="dark">
+                  {value} {fiatUnit}
+                </Label>
+              )}
+            />
+          </RowContainer>
+        )}
       </RowContainer>
-      {fee && (
+      {fee && feeRateUnits && (
```

Comment:
It's really hard to understand what's going on here and also the code that's displaying the fiat rate is duplicated 😅 Can you refactor this to have clearly named boolean values to describe the different situations and extract the duplicate code to a shared var?